packages/jsii-pacmak/lib/targets/java.ts

```diff
@@ -779,53 +795,46 @@ class JavaGenerator extends Generator {
         }
     }
 
-    private emitInterfaceBuilder(ifc: spec.InterfaceType) {
+    private emitDataType(ifc: spec.InterfaceType) {
         const interfaceName = ifc.name;
         const builderName = 'Builder';
 
-        this.code.line();
+        // Start builder()
         this.code.line('/**');
         this.code.line(` * @return a {@link Builder} of {@link ${interfaceName}}`);
         this.code.line(' */');
         this.emitStabilityAnnotations(ifc);
         this.code.openBlock(`static ${builderName} builder()`);
         this.code.line(`return new ${builderName}();`);
         this.code.closeBlock();
+        this.code.line();
+        // End builder()
 
-        interface Prop {
-            docs?: spec.Docs
-            spec: spec.Property
-            propName: string
-            fieldName: string
-            fieldJavaType: string
-            javaTypes: string[]
-            nullable: boolean
-            inherited: boolean
-            immutable: boolean
-        }
-
-        const props = new Array<Prop>();
-
-        // collect all properties from all base structs
+        // collect all properties from all base structs and dedupe by name. It is assumed that the generation of the
+        // assembly will not permit multiple overloaded inherited properties with the same name and that this will be
+        // enforced by Typescript constraints.
+        const propsByName = new Map<string, JavaProp>();
```

Comment:
This could be simplified to be a plain object:

```ts
const propsByName: { [name: string]: JavaProp } = {};
```

This way you can just `propsByName[prop.propName] = prop;`. That'll look more idomatically TypeScript.